R/tax_range_time.R

```diff
@@ -0,0 +1,178 @@
+      df <- palaeoverse::GTS2012
+    }
+    df <- subset(df, rank == "stage")
+    # Which intervals are present?
+    unique_intervals <- unique(c(occdf[, max_ma], occdf[, min_ma]))
+    # Which of these intervals are present in the GTS
+    vec <- which(df$interval_name %in% unique_intervals)
+    # If not available, stop and return error
+    if (length(vec) != length(unique_intervals)) {
+      error_vec <- which(!unique_intervals %in% df$interval_name)
+      if (return_error == TRUE) {
+        return(error_vec)
+      } else {
+      stop(
+        paste(c(
+          "Check spelling of specified max_ma and min_ma intervals.
+  Available interval names are accessible via GTS2020 and GTS2012.
+  Check rows:",
+          capture.output(print(error_vec))
+        ),
+        collapse = "\n"
+      ))
+      }
+    }
+    # Convert character to numeric ages
+    for (i in seq_along(df$interval_name)) {
+      occdf[which(occdf[, max_ma] == df$interval_name[i]), max_ma] <-
+        df$max_ma[i]
+      occdf[which(occdf[, min_ma] == df$interval_name[i]), min_ma] <-
+        df$min_ma[i]
+    }
+    occdf[, max_ma] <- as.numeric(occdf[, max_ma])
+    occdf[, min_ma] <- as.numeric(occdf[, min_ma])
+  }
+  #=== Temporal range ===
+  # Generate dataframe for population
+  temp_df <- data.frame(taxa = unique_taxa,
+                          taxa_id = seq(1, length(unique_taxa), 1),
+                          FAD_ma = rep(NA, length(unique_taxa)),
+                          LAD_ma = rep(NA, length(unique_taxa)),
+                          range_myr = rep(NA, length(unique_taxa)))
+    # Run for loop across unique taxa
+    for(i in seq_along(unique_taxa)){
+      vec <- which(occdf[, name] == unique_taxa[i])
+      temp_df$FAD_ma[i] <- max(occdf[vec, max_ma])
+      temp_df$LAD_ma[i] <- min(occdf[vec, min_ma])
+      temp_df$range_myr[i] <- temp_df$FAD_ma[i] - temp_df$LAD_ma[i]
+    }
+    # Remove row names
+    row.names(temp_df) <- NULL
+    # Round off values
+    temp_df[, c("FAD_ma", "LAD_ma", "range_myr")] <- round(
+      x = temp_df[, c("FAD_ma", "LAD_ma", "range_myr")], digits = 3)
+    # Plot data?
+    if (plot == TRUE){
+      x_range <- c(max(temp_df$FAD_ma), min(temp_df$LAD_ma))
+      y_range <- c(0, nrow(temp_df))
+      plot(x = NA,
+           y = NA,
+           xlim = x_range,
+           ylim = y_range,
+           axes = TRUE,
+           xaxt = "n",
+           xlab = NA,
+           ylab = "Taxa ID",
+           main = "Temporal range of taxa")
+      segments(x0 = temp_df$FAD_ma,
+               x1 = temp_df$LAD_ma,
+               y0 = temp_df$taxa_id,
+               col = temp_df$taxa_id)
+      points(x = temp_df$FAD_ma,
+             y = temp_df$taxa_id,
+             pch = 20,
+             col = temp_df$taxa_id)
+      points(x = temp_df$LAD_ma,
+             y = temp_df$taxa_id,
+             pch = 20,
+             col = temp_df$taxa_id)
+      axis_geo(side = 1, intervals = "periods")
```

Comment:
axis_geo isn't working for me here, although I guess it shouldn't be a problem after merging.

Reply:
I got there eventually, should be fine as long as deeptime and all other functions in palaeoverse are switched on when the package is loaded.